File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06IO2/BuilderAndFabric/BuilderAndFabric/IdBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderAndFabric
{
    class IdBuilder
    {
        private HashSet<long> hs = new HashSet<long>();
        private string Delimeter, IfEmpty;
        public IdBuilder(string Delimeter=", ",string IfEmpty="NULL") { SetDelimeter(Delimeter).SetIfEmpty(IfEmpty); }
        public IdBuilder SetDelimeter(string Delimeter) { this.Delimeter = Delimeter; return this; }
        public IdBuilder SetIfEmpty(string IfEmpty) { this.IfEmpty = IfEmpty; return this; }
        public IdBuilder Clear() { hs.Clear(); return this; }
        public IdBuilder Append(long id) { hs.Add(id); return this; }
        public override string ToString()
        {
            if (hs.Count == 0) return IfEmpty;
            List<long> sl = hs.ToList();
            sl.Sort();
            StringBuilder sb = new StringBuilder();
            /*///
            sb.Append(sl[0]);
            for (int i = 1; i < sl.Count; ++i) sb.Append(Delimeter).Append(sl[i]);
            /*/
            bool first = true;
            foreach(long id in sl)
            {
                if (first) first = false; else sb.Append(Delimeter);
                sb.Append(id);
            }
            //*///
            return sb.ToString();
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2020s/IZ08IO1/ControlBase/src/IdBuilder.java
import java.util.Arrays;
import java.util.HashSet;

public class IdBuilder 
{
    private final HashSet<Long> lst=new HashSet<>();
    private String Delimeter;
    private String IfEmpty;
    public IdBuilder(String Delimeter,String IfEmpty) 
    {
        this.Delimeter=Delimeter;
        this.IfEmpty=IfEmpty;
    }
    public IdBuilder(String Delimeter) { this(Delimeter,"NULL"); }
    public IdBuilder() { this(", "); }
    public IdBuilder setDelimeter(String Delimeter) { this.Delimeter=Delimeter; return this; }
    public IdBuilder setIfEmpty(String IfEmpty) { this.IfEmpty=IfEmpty; return this; }
    public IdBuilder append(long id) { lst.add(id); return this; }
    public IdBuilder clear() { lst.clear(); return this; }
    @Override public String toString() 
    {
        if(lst.size()<=0) return IfEmpty;
        Long[] tb=new Long[lst.size()];
        lst.toArray(tb);
        Arrays.sort(tb);
        final StringBuilder sb=new StringBuilder();
        //*///
        sb.append(tb[0]);
        for(int i=1;i<tb.length;++i) sb.append(Delimeter).append(tb[0]);
        /*/
        boolean first=true;
        for(long id:tb)
        {
            if(first) first=false;
            else sb.append(Delimeter);
            sb.append(id);
        }        
        //*///
        return sb.toString();
    }        
}
